collector/smb_server_shares.go

```diff
@@ -0,0 +1,396 @@
+// +build windows
+
+package collector
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+func init() {
+	registerCollector("smb_server_shares", NewSmbServerSharesCollector, "SMB Server Shares")
```

Comment:
I think maybe we'd like to keep other SMB-related metrics in the same collector, so generalise this to just being the `smb` collector
(Unless there's something I'm missing which makes that a bad idea)

Reply:
sure, I just reflected the naming structure I saw so far
...I'll change it to smb